plugins/urls/handlers/website.py

```diff
@@ -0,0 +1,266 @@
+__author__ = 'Gareth Coles'
+
+import os
+import re
+import socket
+
+from bs4 import BeautifulSoup
+from cookielib import LoadError
+from kitchen.text.converters import to_unicode, to_bytes
+from netaddr import IPAddress
+from twisted.web._newclient import ResponseNeverReceived
+from txrequests import Session
+
+from plugins.urls.handlers.handler import URLHandler
+from plugins.urls.cookiejar import ChocolateCookieJar
+from utils.misc import str_to_regex_flags
+
+
+class WebsiteHandler(URLHandler):
+    name = "website"
+
+    criteria = {
+        "protocol": re.compile(r"http|https", str_to_regex_flags("iu"))
+    }
+
+    group_sessions = {}
+    global_session = None
+
+    cookies_base_path = "data/plugins/urls/cookies"
+
+    def __init__(self, plugin):
+        super(WebsiteHandler, self).__init__(plugin)
+
+        if not os.path.exists(self.cookies_base_path):
+            os.makedirs(self.cookies_base_path)
+
+        if not os.path.exists(self.cookies_base_path + "/groups"):
+            os.makedirs(self.cookies_base_path + "/groups")
+
+        self.reload()
+
+    def call(self, url, context):
+        if self.check_blacklist(url, context):
+            self.urls_plugin.logger.warn(
+                "URL %s is blacklisted, ignoring.." % url.text
+            )
+            return
+
+        try:
+            ip = IPAddress(socket.gethostbyname(url.domain))
+        except Exception as e:
+            context["event"].target.respond(
+                '"{0}" at {1}'.format(e, url.domain)
+            )
+
+            self.plugin.logger.warn(str(e))
+            return False
+
+        if ip.is_loopback() or ip.is_private() or ip.is_link_local():
+            self.plugin.logger.warn("Prevented a port-scan")
+            return False
+
+        headers = {}
+
+        if url.domain in context["config"]["spoofing"]:
+            user_agent = context["config"]["spoofing"][url.domain]
+
+            if user_agent:
+                headers["User-Agent"] = user_agent
+        else:
+            headers["User-Agent"] = ("Mozilla/5.0 (Windows NT 6.3; rv:36.0) "
```

Comment:
There should probably be an optional config value for this, similar to Accept-Language.